hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -70,18 +69,12 @@ protected long mintFungible(
             @NonNull final Token token,
             @NonNull final TokenRelation treasuryRel,
             final long amount,
-            final boolean isMintOnTokenCreation,
             @NonNull final WritableAccountStore accountStore,
             @NonNull final WritableTokenStore tokenStore,
             @NonNull final WritableTokenRelationStore tokenRelationStore) {
         requireNonNull(token);
         requireNonNull(treasuryRel);
-        // validate token supply key exists for mint or burn.
-        // But this flag is not set when mint is called on token creation with initial supply.
-        // We don't need to check the supply key ONLY in that case
-        if (!isMintOnTokenCreation) {
```

Comment:
⚠️ I don't understand how we can safely delete this...won't it break any `@HapiTest` that creates a token without a supply key?

Reply:
Hey @tinker-michaelj, the `BaseTokenHandler::mintFungible` is only called from 2 places: `TokenCreateHandler` where the `isMintOnTokenCreation` flag is hard-coded to true and from `TokenMintHandler`, where the flag is false. So the supply key check is only executed on mints. Thats why I removed the logic from the `BaseTokenHandler` and moved it in the `TokenMintHandler`

Reply:
Ah makes sense. 🙏